internal/cmd/grpc.go

```diff
@@ -154,119 +144,12 @@ func NewGRPCServer(
-			git.WithRef(cfg.Storage.Git.Ref),
-			git.WithPollInterval(cfg.Storage.Git.PollInterval),
-			git.WithInsecureTLS(cfg.Storage.Git.InsecureSkipTLS),
-		}
-
-		if cfg.Storage.Git.CaCertBytes != "" {
-			opts = append(opts, git.WithCABundle([]byte(cfg.Storage.Git.CaCertBytes)))
-		} else if cfg.Storage.Git.CaCertPath != "" {
-			if bytes, err := os.ReadFile(cfg.Storage.Git.CaCertPath); err == nil {
-				opts = append(opts, git.WithCABundle(bytes))
-			} else {
-				return nil, err
-			}
-		}
-
-		auth := cfg.Storage.Git.Authentication
-		switch {
-		case auth.BasicAuth != nil:
-			opts = append(opts, git.WithAuth(&http.BasicAuth{
-				Username: auth.BasicAuth.Username,
-				Password: auth.BasicAuth.Password,
-			}))
-		case auth.TokenAuth != nil:
-			opts = append(opts, git.WithAuth(&http.TokenAuth{
-				Token: auth.TokenAuth.AccessToken,
-			}))
-		case auth.SSHAuth != nil:
-			var method *gitssh.PublicKeys
-			if auth.SSHAuth.PrivateKeyBytes != "" {
-				method, err = gitssh.NewPublicKeys(
-					auth.SSHAuth.User,
-					[]byte(auth.SSHAuth.PrivateKeyBytes),
-					auth.SSHAuth.Password,
-				)
-			} else {
-				method, err = gitssh.NewPublicKeysFromFile(
-					auth.SSHAuth.User,
-					auth.SSHAuth.PrivateKeyPath,
-					auth.SSHAuth.Password,
-				)
-			}
-			if err != nil {
-				return nil, err
-			}
-
-			// we're protecting against this explicitly so we can disable
-			// the gosec linting rule
-			if auth.SSHAuth.InsecureIgnoreHostKey {
-				// nolint:gosec
-				method.HostKeyCallback = ssh.InsecureIgnoreHostKey()
-			}
-
-			opts = append(opts, git.WithAuth(method))
-		}
-
-		source, err := git.NewSource(logger, cfg.Storage.Git.Repository, opts...)
-		if err != nil {
-			return nil, err
-		}
-
-		store, err = fs.NewStore(logger, source)
-		if err != nil {
-			return nil, err
-		}
-	case config.LocalStorageType:
-		source, err := local.NewSource(logger, cfg.Storage.Local.Path)
-		if err != nil {
-			return nil, err
-		}
-
-		store, err = fs.NewStore(logger, source)
-		if err != nil {
-			return nil, err
-		}
-	case config.ObjectStorageType:
-		store, err = NewObjectStore(cfg, logger)
-		if err != nil {
-			return nil, err
-		}
-	case config.OCIStorageType:
-		var opts []containers.Option[oci.StoreOptions]
-		if auth := cfg.Storage.OCI.Authentication; auth != nil {
-			opts = append(opts, oci.WithCredentials(
-				auth.Username,
-				auth.Password,
-			))
-		}
-
-		ocistore, err := oci.NewStore(logger, cfg.Storage.OCI.BundlesDirectory, opts...)
-		if err != nil {
-			return nil, err
-		}
-
-		ref, err := oci.ParseReference(cfg.Storage.OCI.Repository)
-		if err != nil {
-			return nil, err
-		}
-
-		source, err := storageoci.NewSource(logger, ocistore, ref,
-			storageoci.WithPollInterval(cfg.Storage.OCI.PollInterval),
-		)
-		if err != nil {
-			return nil, err
-		}
-
-		store, err = fs.NewStore(logger, source)
+	default:
+		// otherwise, attempt to configure a declarative backend store
+		store, err = fsstore.NewStore(ctx, logger, cfg)
```

Comment:
![](https://media.giphy.com/media/7FyMQm2vBiTjG/giphy.gif)

Reply:
Just don't open that new package and it's like all that code doesn't exist.